packages/cubejs-clickhouse-driver/driver/ClickHouseDriver.js

```diff
@@ -20,7 +22,14 @@ class ClickHouseDriver extends BaseDriver {
       create: async () => new ClickHouse({
         ...this.config,
         queryOptions: {
-          join_use_nulls: 1,
+          //
+          //
+          // If ClickHouse user's permissions are restricted with "readonly = 1",
+          // change settings queries are not allowed. Thus, "join_use_nulls" setting
+          // can not be changed
+          //
+          //
+          ...(this.config.readonly ? {} : { join_use_nulls: 1 }),
```

Comment:
@igorlukanin I think there is a misunderstanding about read-only flag for connections and `readOnly` option.

inside Cube.js option called `readOnly` is used to choose another strategy for  pre-aggregation creation. (pre-agg will be done without creating table and type detection will be done from db's response which has limitations - https://cube.dev/docs/pre-aggregations#read-only-data-source-pre-aggregations-limitations)

If you are trying to get another pre-aggregation strategy ☝️ , you should implement readOnly method

```js
  readOnly() {
    return !!this.config.readOnly;
  }
```

And in this case I think we should not do read-only connection.

Thanks

Reply:
Sure, you're right. I'm not talking about pre-aggregations here.